rmf_traffic_ros2/src/rmf_traffic_ros2/schedule/MirrorManager.cpp

```diff
@@ -282,11 +282,11 @@ class MirrorManager::Implementation
       register_query_request.query = convert(query);
       register_query_client->async_send_request(
         std::make_shared<RegisterQuery::Request>(register_query_request),
-        [&](const RegisterQueryFuture response)
+        [this](const RegisterQueryFuture response)
```

Comment:
Just to be clear, this is still effectively capture by reference in the sense that `this` is a raw pointer, making it equivalent to an unsafe reference. The original version of the code is functionally equivalent, because `this` pointer is always captured by value, and `this->` is always implied when using a member variable/function within a lambda.

Reply:
What would be the correct way to safely capture here?

Reply:
There are two options for safe capturing:
1. Every member field that needs to be used in the callback should be stored as a `std::shared_ptr<T>` and each relevant `std::shared_ptr<T>` should be copied into the capture list.
2. The whole `Implementation` class should have a nested `Shared` class that contains all the fields, and you should copy a `std::weak_ptr<Shared>` into the capture list. You can find an example of this in [`rmf_traffic::schedule::Participant::Implementation::Shared`](https://github.com/open-rmf/rmf_traffic/blob/e54c63ea75c20971be866a9fc155ee171c17e8fb/rmf_traffic/src/rmf_traffic/schedule/internal_Participant.hpp#L40-L77).